tests/net/lib/coap/src/main.c

```diff
@@ -1771,4 +1791,45 @@ ZTEST(coap, test_transmission_parameters)
 	zassert_equal(pending->params.max_retransmission, 2, "Wrong max retransmission value");
 }
 
+ZTEST(coap, test_notify_age)
+{
+	uint8_t valid_request_pdu[] = {
+		0x45, 0x01, 0x12, 0x34, 't', 'o', 'k', 'e', 'n', 0x60, /* enable observe option */
+		0x51, 's',  0x01, '2',                                 /* path */
+	};
+
+	struct coap_packet req;
+	struct coap_option options[4] = {};
+	uint8_t *data = data_buf[0];
+	uint8_t opt_num = ARRAY_SIZE(options) - 1;
+	struct coap_resource *resource = &server_resources[1];
+	int r;
+
+	memcpy(data, valid_request_pdu, sizeof(valid_request_pdu));
+
+	r = coap_packet_parse(&req, data, sizeof(valid_request_pdu), options, opt_num);
+	zassert_equal(r, 0, "Could not initialize packet");
+
+	r = coap_handle_request(&req, server_resources, options, opt_num,
+				(struct sockaddr *)&dummy_addr, sizeof(dummy_addr));
+	zassert_equal(r, 0, "Could not handle packet");
+
+	// Forward time a bit, as not to run this 8 million time
+	resource->age = INT32_MAX - 10;
```

Comment:
Is INT32_MAX actually correct assumption here, we've just introduced code that prevented 24-bit age overflow. Perhaps we should introduce a define for maximum age value and use it in code and test here?